zed-editor: 0.133.7 -> 0.134.4

GaetanLepage opened the pull request:
## Description of changes

Changelog: https://github.com/zed-industries/zed/releases/tag/v0.134.4

cc @niklaskorz 

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

GaetanLepage commented:
Result of `nixpkgs-review pr 310202` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>zed-editor</li>
  </ul>
</details>

GaetanLepage commented:
Result of `nixpkgs-review pr 310202` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>zed-editor</li>
  </ul>
</details>

GaetanLepage commented:
Result of `nixpkgs-review pr 310202` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>zed-editor</li>
  </ul>
</details>

jansol reviewed (changes requested):
There is now [an official .desktop file](https://github.com/zed-industries/zed/blob/v0.134.3/crates/zed/resources/zed.desktop), we should use that instead of our own. Note that it should still be called `dev.zed.Zed.desktop`, both according to XDG spec and [upstream install script behaviour](https://github.com/zed-industries/zed/pull/11337/files)

Aside from that, LGTM.
The blade-graphics hash change is expected as that is a git dependency referred to by commit hash and it has been updated recently.

GaetanLepage committed:
zed-editor: 0.133.7 -> 0.134.4

Diff: https://github.com/zed-industries/zed/compare/refs/tags/v0.133.7...0.134.4

Changelog: https://github.com/zed-industries/zed/releases/tag/v0.134.4

jansol commented on pkgs/by-name/ze/zed-editor/package.nix:
```diff
@@ -139,48 +139,9 @@ rustPlatform.buildRustPackage rec {
     mv $out/bin/Zed $out/bin/zed
     install -D ${src}/crates/zed/resources/app-icon@2x.png $out/share/icons/hicolor/1024x1024@2x/apps/Zed.png
     install -D ${src}/crates/zed/resources/app-icon.png $out/share/icons/hicolor/512x512/apps/Zed.png
```
```suggestion
    install -D ${src}/crates/zed/resources/app-icon@2x.png $out/share/icons/hicolor/1024x1024@2x/apps/zed.png
    install -D ${src}/crates/zed/resources/app-icon.png $out/share/icons/hicolor/512x512/apps/zed.png
```

GaetanLepage replied:
Good catch ! Done.

GaetanLepage commented:
Result of `nixpkgs-review pr 310202` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>zed-editor</li>
  </ul>
</details>

GaetanLepage commented:
Result of `nixpkgs-review pr 310202` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>zed-editor</li>
  </ul>
</details>

jansol commented on pkgs/by-name/ze/zed-editor/package.nix:
```diff
@@ -137,50 +136,11 @@ rustPlatform.buildRustPackage rec {
-    install -D ${src}/crates/zed/resources/app-icon.png $out/share/icons/hicolor/512x512/apps/Zed.png
+    install -D ${src}/crates/zed/resources/app-icon@2x.png $out/share/icons/hicolor/1024x1024@2x/apps/zed.png
+    install -D ${src}/crates/zed/resources/app-icon.png $out/share/icons/hicolor/512x512/apps/zed.png
+    install -Dm644 ${src}/crates/zed/resources/zed.desktop -t $out/share/applications/dev.zed.Zed.desktop
```
Something went wrong with the desktop file install location...
```
$ ls --tree result
result ⇒ /nix/store/yymqkjsc84d3bk4bfn3zrz99c687y989-zed-0.134.4
├── bin
│   └── zed
└── share
    ├── applications
    │   └── dev.zed.Zed.desktop
    │       └── zed.desktop
```

GaetanLepage replied:
Indeed, any idea why it behaves like this ?

GaetanLepage replied:
Seems good now.
The `-t` option of `install` means "target directory".
